cmd/bootstrap/cmd/finalize.go

```diff
@@ -466,7 +466,7 @@ func readDKGData() dkg.DKGData {
 
 	dkgData := dkg.DKGData{
 		PrivKeyShares: nil,
-		PubGroupKey:   encodableDKG.GroupKey,
+		PubGroupKey:   encodableDKG.GroupKey.PublicKey,
```

Comment:
I've did this and we absolutely need this since there is a cast somewhere in crypto layer to `(*PubKeyBLSBLS12381)` which is not doable if we are dealing with `encodable.RandomBeaconPubKey`

Reply:
https://github.com/onflow/flow-go/blob/feature/consensus-voting/crypto/bls_thresholdsign.go#L167

One way of solving this is to change the check, to check the `Algorithm()`

Reply:
@tarakby

Reply:
I see 🤔 
I used to make these checks in the crypto library using `Algorithm()` but I changed them to type assertions for a long reason. (`Algorithm()` works well in the case of threshold signatures though). 

I'll create a ticket to fix these issues  in the crypto lib. 

Meanwhile, is there a kind of recursive type assertion in Go? something like `error.As` for errors ? @durkmurder